packages/nx-sonarqube/src/executors/scan/utils/utils.ts

```diff
@@ -103,9 +103,31 @@ export async function determinePaths(
   });
 
   deps.workspaceLibraries
-    .filter((project) =>
-      options.skipImplicitDeps ? project.type !== DependencyType.implicit : true
+    .filter(
+      (project) =>
+        context.projectName === project.name ||
+        (options.skipPaths?.length
+          ? options.skipPaths.some((path) => project.sourceRoot.includes(path))
+          : true)
     )
+    .filter(
+      (project) =>
+        context.projectName === project.name ||
+        (options.skipProjects?.length
+          ? options.skipProjects.includes(project.name)
+          : true)
+    )
+    .filter((project) => {
+      if (
+        context.projectName === project.name ||
+        !options.skipDependencyTypes?.length
+      ) {
+        return true;
+      }
+      return options.skipDependencyTypes.includes(
```

Comment:
I tested this logic filter in my own repo, and it does not work. It needs to filter by DependencyType.static|implicit|dynamic based on the options array.

```json
"skipDependencyTypes": ["static"]
```

```sh
> nx run lib-b:sonar

Included sources paths: libs/lib-a/src,libs/lib-b/src
Included lcov paths: coverage/libs/lib-a/lcov.info,coverage/libs/lib-b/lcov.info
```